src/common.c

```diff
@@ -305,3 +305,25 @@ void raftNodeIdToString(char *output, const char *dbid, raft_node_id_t raft_id)
 {
     snprintf(output, RAFT_SHARDGROUP_NODEID_LEN + 1, "%.32s%08x", dbid, raft_id);
 }
+
+int validateCommandACL(RedisModuleCtx *ctx, RaftRedisCommand *cmd)
+{
+    RedisModuleString *user_name = RedisModule_GetCurrentUserName(ctx);
+    RedisModuleUser *user = RedisModule_GetModuleUserFromUserName(user_name);
+    int ret = RedisModule_ACLCheckCommandPermissions(user, cmd->argv, cmd->argc);
```

Comment:
Can we use dry run mode here?

Reply:
modified multi.c to use dry run mode (though only for ACL, not for OOM, oom is different in multi, i.e. even non deny oom commands are rejected when OOM (as dont want multi queue to grow without bounds)